pkg/yamltemplate/filetests/ytt-library/ip/parse-addr.yml

```
@@ -0,0 +1,22 @@
+#@ load("@ytt:ip", "ip")
+
+ipv4:
+  ip: #@ ip.parse_addr("192.0.2.1")
```

```suggestion
  ip: #@ ip.parse_addr("192.0.2.1").string()
```

We established in #372 the convention of _not_ assuming how to encode the given value. In the current, we're presuming to encode as a string.
For now, we're preserving that decision by making custom types non-convertible by default, but also provide the string encoding (i.e. `string()` method).

Getting there means `AddrValue` satisfies:
https://github.com/vmware-tanzu/carvel-ytt/blob/214258be2519cdb66953c8ab2e3e234f879af8ca/pkg/template/core/starlark_value.go#L19-L21

... and does _not_ satisfy:

https://github.com/vmware-tanzu/carvel-ytt/blob/214258be2519cdb66953c8ab2e3e234f879af8ca/pkg/template/core/starlark_value.go#L13-L15

... and then providing a `string()` method on it.

(see `URLValue` and `URLUser`: https://github.com/vmware-tanzu/carvel-ytt/pull/372/files?file-filters%5B%5D=.go#diff-526a88c3f59e3e4dc6e4124408bee316ab1ddbb86001953e222fcd01e603280b )

---

fixed